### PR TITLE
Update docs for the new runner

### PR DESCRIPTION
**Pull request by apache8080**

### CODE REVIEW

File: docs/writing_your_plugin.md [hunk 1]
```diff
@@ -140,19 +160,19 @@ functionality this way.
...
+    
             virtual void _p_one_iteration override() {
                 std::cout << "Running" << std::endl;
                 auto target = std::chrono::high_resolution_clock::now() +  std::chrono::milliseconds{10};
                 reliable_sleep(target);
             }
-
+    
         private:
             const std::shared_ptr<switchboard> pb;
```
Comment (charmoniumQ) [inline docs/writing_your_plugin.md:171]:
```suggestion
            const std::shared_ptr<switchboard> sb;
```
Just noticed this small bug.

File: mkdocs.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ nav:
     - 'Default Plugins': default_plugins.md
     - 'Writing Your Plugin': writing_your_plugin.md
     - 'Building ILLIXR': building_ILLIXR.md
+    - 'ILLIXR Debugging Tips': debugging_ILLIXR.md
```
Comment (charmoniumQ) [inline mkdocs.yml:11]: Is this supposed to point to an empty file?

**Review by charmoniumQ — APPROVED** [review]

File: mkdocs.yml (hunk 2 above)
Comment (apache8080) [reply mkdocs.yml:11]: That was a mistake.

File: docs/debugging_illixr.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# ILLIXR Debugging Tips
+
+## Debugging Locally
+The config described in [Building ILLIXR][1] supports running the runtime with arbitrary commands like `gdb`. When debuggin locally we recommend using either `gdb` or `valgrind` in this way.
```
Comment (mhuzai) [inline docs/debugging_illixr.md:4]: debugging.

File: docs/debugging_illixr.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## Debugging PRs
+### Running the CI/CD Docker Container Locally
+Follow these steps when a CI/CD build fails on a PR:
+- Click details on the failing build
+- In the build view go to the Push Docker Image tab and copy the `docker push ghcr.io/illixr/illixr-tests:<branch-name>`
+- Then in your terminal run `docker pull ghcr.io/illixr/illixr-tests:<branch-name>`
+- Finally run  `docker run -it --entrypoint /bin/bash ghcr.io/illixr/illixr-tests:<branch-name>`
```
Comment (mhuzai) [inline docs/debugging_illixr.md:12]: Remove extra space after run.

File: docs/debugging_illixr.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# ILLIXR Debugging Tips
+
+## Debugging Locally
+The config described in [Building ILLIXR][1] supports running the runtime with arbitrary commands like `gdb`. When debuggin locally we recommend using either `gdb` or `valgrind` in this way.
+
+## Debugging PRs
+### Running the CI/CD Docker Container Locally
+Follow these steps when a CI/CD build fails on a PR:
+- Click details on the failing build
+- In the build view go to the Push Docker Image tab and copy the `docker push ghcr.io/illixr/illixr-tests:<branch-name>`
```
Comment (mhuzai) [inline docs/debugging_illixr.md:10]: copy the `docker...` *command*.

File: docs/writing_your_plugin.md [hunk 6]
```diff
@@ -77,41 +72,66 @@ functionality this way.
...
+     # command: gdb -q --args %a
+   profile: opt
+   ```
+
+   
+
+5. Finally, run ILLIXR with your new plugin with the following command: `./runner.sh configs/native.yaml`
+
+6. This is all that is required to be a plugin which can be loaded by Spindle in the ILLIXR
+   runtime. Reading and writing from Phonebook and Switchboard optional, but nearly every plugin
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:109]: Switchboard *is* optional

File: mkdocs.yml (hunk 2 above)
Comment (mhuzai) [reply mkdocs.yml:12]: debugging_illixr.md?
Also, please change building_ILLIXR to building_illixr -- I forgot to do it in another PR.

**Review by mhuzai — COMMENTED** [review]
Modifying a plugin is identical to writing your plugin.

**Comment by apache8080** [comment]
> Modifying a plugin is identical to writing your plugin.

I made them separate since it is slightly different steps and I wanted to have the distinction

**Review by charmoniumQ — DISMISSED** [review]

File: docs/writing_your_plugin.md [hunk 7]
```diff
@@ -77,41 +72,66 @@ functionality this way.
...
-
-5.  This is all that is required to be a plugin which can be loaded by Spindle in the ILLIXR
-    runtime. Reading and writing from Phonebook and Switchboard optional, but nearly every plugin
-    does it. See `default_plugins.md` for more details.
-
-    First, we can query the [`phonebook`][2] to get various services including [`switchboard`][5]. Then we
-    query [`switchboard`][5] for event-streams (topics). We will read `topic1`, write to `topic2`, and
-    schedule computation on `topic 3`. See the API documentation for `phonebook` and `switchboard`
-    for more details.
+4. At this point, you should be able to build your plugin with ILLIXR. Move to the ILLIXR repo and update `configs/native.yaml`. See [Building ILLIXR][1] for more details on the config file.
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:80]: Need some info on what to do when the new plugin replaces an old one. For example, can't have two SLAMs running at the same time and publishing to the same topic.

File: docs/modifying_a_plugin.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+           git_repo: https://github.com/ILLIXR/audio_pipeline.git
+           version: 3433bb452b2ec661c9d3ef65d9cf3a2805e94cdc
+     ```
+
+   - New Config
+
+     ```
+     plugin_group:
+       - path: timewarp_gl/
+       - name: /PATH/TO/LOCAL/PLUGIN
```
Comment (mhuzai) [inline docs/modifying_a_plugin.md:27]: Maybe make this `/PATH/TO/LOCAL/AUDIO-PLUGIN` given that the example clones audio and replaces audio.

File: docs/modifying_a_plugin.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# Modifying a plugin
+
+## Tutorial
+
+This is how you can modify an existing ILLIXR plugin
+
+1. Clone the repository for the component you want to modify. e.g: `git clone https://github.com/ILLIXR/audio_pipeline.git`
+
+2.  Modify the config file like this:
```
Comment (mhuzai) [inline docs/modifying_a_plugin.md:9]: Step 2 should be "make changes..." and then step 3 should be "modify config file".

For the sake of completeness and clarity, maybe step 4 to say "now when you run ILLIXR, the local audio component gets built and run".

And maybe there should be a step 5 as well describing what to do when you push your changes back up.

**Review by mhuzai — COMMENTED** [review]
Left some comments.

Also, question about dev process with new runner, @charmoniumQ and @apache8080: local plugins will require a manual symlink to common, which must then be removed before pushing up, yes? If so, this could be very confusing for people.

File: docs/modifying_a_plugin.md (hunk 9 above)
Comment (apache8080) [reply docs/modifying_a_plugin.md:9]: Are step 2 and step 4 necessary? Its sort of implied.

**Review by mhuzai — DISMISSED** [review]

**Review by mhuzai — APPROVED** [review]

**Review by charmoniumQ — APPROVED** [review]

**Pull request merged**